test/sphere_Plm.jl

```diff
@@ -0,0 +1,143 @@
+function legendre(n, x)
```

Comment:
I'll take a look at this package. It is the optimal if we could directly use this package.

Reply:
https://juliapackages.com/p/sphericalharmonics?

Reply:
> Swarztrauber, P. N. (1993). The Vector Harmonic Transform Method for Solving Partial Differential Equations in Spherical Geometry. Monthly Weather Review, 121(12), 3415–3437.

Reply:
@simonbyrne  I tried both Oswald's function and the Plm function from0` SphericalHarmonics.jl`. It looks like they are different.

Reply:
They might be off just by some factor. I am not very sure.

Reply:
Could this just be because different weighting conventions are used in calculating the Legendre polynomials? Apparently there is no one set method (Ehrendorfer 92 book). I agree that a tested and maintained package may be the way forward.

Reply:
@simonbyrne  I have tested the Plm function in the pkg you mentioned above. It is not producing what Oswald's function produces. Also, I compared the Plm results to an online calculator https://keisan.casio.com/exec/system/1180573406, Oswald's function is consistent with the online tool but the pkg's function gives quite odd outputs. I suggest we use Oswald's code.

Reply:
@jiahe23 Okay, that's fine. Can you just add a comment with a reference to the paper, so that we know where it comes from?

Reply:
I asked about registering the Legendre package and the author seemed receptive: https://github.com/jmert/Legendre.jl/issues/28

Reply:
@jakebolewski  Thanks, Jake. Meanwhile, I will add his personal registry on my local machine and test if I could reproduce the VSH from his Polynomial function.